pennylane/compiler/__init__.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+.. currentmodule:: pennylane
+
+This module provides support for hybrid quantum-classical compilation.
+Through the use of the :func:`~.qjit` decorator, entire workflows
+can be just-in-time (JIT) compiled --- including both quantum and
+classical processing --- down to a machine binary on first
+function execution. Subsequent calls to the compiled function will execute
+the previously compiled binary, resulting in significant
+performance improvements.
+
+Currently, PennyLane supports the
+`Catalyst <https://github.com/pennylaneai/catalyst>`__ hybrid compiler
+with the :func:`~.qjit` decorator. A significant benefit of Catalyst
+is the ability to preserve complex control flow around quantum
+operations — such as if statements and for loops, and including measurement
+feedback — during compilation, while continuing to support end-to-end
+autodifferentiation.
+
+.. note::
+
+    Catalyst currently only supports the JAX interface of PennyLane.
+
+Overview
+--------
+
+The main entry point to hybrid compilation in PennyLane
+is via the qjit decorator.
+
+.. autosummary::
+    :toctree: api
+
+    ~qjit
+
+In addition, several developer functions are available to probe
+available hybrid compilers.
+
+.. autosummary::
+    :toctree: api
+
+    ~compiler.available_compilers
+    ~compiler.available
+    ~compiler.active_compiler
+    ~compiler.active
+
+Compiler
+--------
+
+The compiler module provides the infrastructure to integrate external
+hybrid quantum-classical compilers with PennyLane, but does not provide
+a built-in compiler.
+
+Currently, only the `Catalyst <https://github.com/pennylaneai/catalyst>`__
+hybrid compiler is supported with PennyLane, however there are plans
+to incorporate additional compilers in the near future.
+
+.. note::
+
+    Catalyst is officially supported on Linux (x86_64) and macOS (aarch64) platforms.
```

Comment:
```suggestion
    Catalyst is officially supported on Linux (``x86_64``) and macOS (``aarch64``) platforms.
```

Reply:
Hmm.. @josh146 these arch types are not Code. Would you still have this suggestion?

Reply:
Good call, lets leave it as is!